cmake/OpenCVFindJPEGXL.cmake

```diff
@@ -0,0 +1,148 @@
+# The script is taken from https://webkit.googlesource.com/WebKit/+/master/Source/cmake/FindJPEGXL.cmake
+
+# Copyright (C) 2021 Sony Interactive Entertainment Inc.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+#[=======================================================================[.rst:
+FindJPEGXL
+---------
+Find JPEGXL headers and libraries.
+Imported Targets
+^^^^^^^^^^^^^^^^
+``JPEGXL::jxl``
+  The JPEGXL library, if found.
+Result Variables
+^^^^^^^^^^^^^^^^
+This will define the following variables in your project:
+``JPEGXL_FOUND``
+  true if (the requested version of) JPEGXL is available.
+``JPEGXL_VERSION``
+  the version of JPEGXL.
+``JPEGXL_LIBRARIES``
+  the libraries to link against to use JPEGXL.
+``JPEGXL_INCLUDE_DIRS``
+  where to find the JPEGXL headers.
+``JPEGXL_COMPILE_OPTIONS``
+  this should be passed to target_compile_options(), if the
+  target is not used for linking
+#]=======================================================================]
+
+if(NOT OPENCV_SKIP_JPEGXL_FIND_PACKAGE)
+find_package(PkgConfig QUIET)
```

Comment:
There is the same call in the root cmake-file: https://github.com/opencv/opencv/blob/6873bdee70528ba6101bd5f4091b5f5302ee70c9/CMakeLists.txt#L710
We don't need to repeat it here.